tasks/retrieval_deal.go

```diff
@@ -17,18 +18,49 @@ type RetrievalTask struct {
 	CARExport  bool   `json:"car_export"`
 }
 
-func MakeRetrievalDeal(ctx context.Context, config NodeConfig, node api.FullNode, task RetrievalTask, log UpdateStatus) error {
-	payloadCid, err := cid.Parse(task.PayloadCID)
+func MakeRetrievalDeal(ctx context.Context, config NodeConfig, node api.FullNode, task RetrievalTask, updateStage UpdateStage, log LogStatus) error {
+	de := &retrievalDealExecutor{
+		dealExecutor: dealExecutor{
+			ctx:    ctx,
+			config: config,
+			node:   node,
+			miner:  task.Miner,
+			log:    log,
+		},
+		task: task,
+	}
+
+	err := executeStage("MinerOnline", updateStage, []step{
+		{de.getTipSet, "Tipset successfully fetched"},
+		{de.getMinerInfo, "Miner Info successfully fetched"},
+		{de.getPeerAddr, "Miner address validated"},
+		{de.netConnect, "Connected to miner"},
```

Comment:
What is the reason for the hierarchy of steps within a stage, instead of having each step as a separate stage?  Would it might be simpler for a retrieval deal to have the following stages:
- GetTipSet
- GetMinerInfo
- GetPeerAddr
- NewConnect
- QueryOffer

Reply:
the stages here were set by Mike's desires for how we mark things in the dashboards

Reply:
so essentially, the fine grained-ness i'm trying to get to is exactly what's outlined here: https://github.com/filecoin-project/dealbot/issues/62